js/src/workspaces/slot.js

```diff
@@ -236,7 +235,20 @@
         });
     },
 
-    clearSlot: function() {
+    destroyEvents:function(){
+      var _this = this;
+      this.events.forEach(function(event){
+        _this.eventEmitter.unsubscribe(event.name,event.handler);
+      });
+    },
+
+    clearSlot: function () {
+      this.destroyEvents();
+
+      // @TODO This is not a task that slots should fulfil
```

Comment:
@aeschylus Please check this, I am not really into what 

>  // This prevents the save controller
 +      // from attempting to re-save the window
 +      // after having already removed it.

it actually fixes